File: src/main/java/chat/teco/tecochat/auth/domain/Authenticator.java
package chat.teco.tecochat.auth.domain;

import static chat.teco.tecochat.auth.exception.AuthenticationExceptionType.NO_REGISTERED_MEMBER;
import static chat.teco.tecochat.common.util.Base64Util.decodeBase64;

import chat.teco.tecochat.auth.exception.AuthenticationException;
import chat.teco.tecochat.member.domain.Member;
import chat.teco.tecochat.member.domain.MemberRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Component;

@RequiredArgsConstructor
@Component
public class Authenticator {

    private final MemberRepository memberRepository;

    public Member authenticateWithBase64(String encodedName) {
        return memberRepository.findByName(decodeBase64(encodedName))
                .orElseThrow(() -> new AuthenticationException(NO_REGISTERED_MEMBER));
    }
}

File: src/main/kotlin/chat/teco/tecochat/domain/auth/Authenticator.kt
package chat.teco.tecochat.domain.auth

import chat.teco.tecochat.domain.member.Member
import chat.teco.tecochat.domain.member.MemberRepository
import chat.teco.tecochat.support.util.Base64Decoder
import org.springframework.stereotype.Component

@Component
class Authenticator(
    private val memberRepository: MemberRepository,
) {

    fun authenticateWithBase64(encodedName: String): Member {
        return memberRepository.findByName(Base64Decoder(encodedName))
            ?: throw IllegalArgumentException("인증에 실패했습니다.")
    }
}
